processing/src/test/java/io/druid/query/DruidProcessingConfigTest.java

```diff
@@ -42,7 +42,9 @@ public void testDeserialization() throws Exception
 
     Assert.assertEquals(1024 * 1024 * 1024, config.intermediateComputeSizeBytes());
     Assert.assertEquals(Integer.MAX_VALUE, config.poolCacheMaxCount());
-    Assert.assertTrue(config.getNumThreads() < Runtime.getRuntime().availableProcessors());
+    if (Runtime.getRuntime().availableProcessors() > 1) {
+      Assert.assertTrue(config.getNumThreads() < Runtime.getRuntime().availableProcessors());
+    }
```

Comment:
it would be nice to also assert that numThreads is 1 when available processors is 1